token-swap/program/src/processor.rs

```diff
@@ -107,26 +107,28 @@ impl State {
     }
 
     /// Issue a spl_token `MintTo` instruction.
-    pub fn token_mint_to(
-        accounts: &[AccountInfo],
-        token_program_id: &Pubkey,
+    pub fn token_mint_to<'a>(
         swap: &Pubkey,
-        mint: &Pubkey,
-        destination: &Pubkey,
-        authority: &Pubkey,
+        token_program: AccountInfo<'a>,
+        mint: AccountInfo<'a>,
+        destination: AccountInfo<'a>,
+        authority: AccountInfo<'a>,
+        nonce: u8,
         amount: u64,
     ) -> Result<(), ProgramError> {
+        let nonce = &[nonce];
         let swap_bytes = swap.to_bytes();
-        let signers = &[&[&swap_bytes[..32]][..]];
+        let signers = &[&[&swap_bytes[..32], nonce][..]];
```

Comment:
So many brackets! I realize that predates your changes, but it might be a little easier to follow if we declare some of it separately. Wdyt?

Reply:
I thought it was a stylistic choice in the code :sweat_smile: it's tough since `invoke_signed` takes `&[&[&[u8]]]` -- I'll clarify this with one more line though, great idea

Reply:
I like the change that you made on L631. Can you do something similar here?

Reply:
Oops, sorry for missing that on the second pass, thanks for spotting it